Userland/DevTools/HackStudio/ToDoEntriesWidget.cpp

```diff
@@ -0,0 +1,114 @@
+
+class ToDoEntriesModel final : public GUI::Model {
+public:
+    enum Column {
+        Filename,
+        Text,
+        Line,
+        Column,
+        __Count
+    };
+
+    explicit ToDoEntriesModel(const Vector<Cpp::TodoEntry>&& matches)
+        : m_matches(move(matches))
+    {
+    }
+
+    virtual int row_count(const GUI::ModelIndex& = GUI::ModelIndex()) const override { return m_matches.size(); }
+    virtual int column_count(const GUI::ModelIndex& = GUI::ModelIndex()) const override { return Column::__Count; }
+
+    virtual String column_name(int column) const override
+    {
+        switch (column) {
+        case Column::Filename:
+            return "Filename";
+        case Column::Text:
+            return "Text";
+        case Column::Line:
+            return "Line";
+        case Column::Column:
+            return "Col";
+        default:
+            VERIFY_NOT_REACHED();
+        }
+    }
+
+    virtual GUI::Variant data(const GUI::ModelIndex& index, GUI::ModelRole role) const override
+    {
+        if (role == GUI::ModelRole::TextAlignment)
+            return Gfx::TextAlignment::CenterLeft;
+        if (role == GUI::ModelRole::Font) {
+            if (index.column() == Column::Text)
+                return Gfx::FontDatabase::default_fixed_width_font();
+            return {};
+        }
+        if (role == GUI::ModelRole::Display) {
+            auto& match = m_matches.at(index.row());
+            switch (index.column()) {
+            case Column::Filename:
+                return match.filename;
+            case Column::Text:
+                return match.content;
+            case Column::Line:
+                return String::formatted("{}", match.line + 1);
+            case Column::Column:
+                return String::formatted("{}", match.column);
+            }
+        }
+        return {};
+    }
+
+    virtual void update() override { }
+    virtual GUI::ModelIndex index(int row, int column = 0, const GUI::ModelIndex& = GUI::ModelIndex()) const override
+    {
+        if (row < 0 || row >= (int)m_matches.size())
+            return {};
+        if (column < 0 || column >= Column::__Count)
+            return {};
+        return create_index(row, column, &m_matches.at(row));
+    }
+
+private:
+    Vector<Cpp::TodoEntry> m_matches;
+};
+
+void ToDoEntriesWidget::refresh()
+{
+    const auto& entries = ToDoEntries::the().get_entries();
+    auto results_model = adopt_ref(*new ToDoEntriesModel(move(entries)));
+    m_result_view->set_model(results_model);
+}
+
+ToDoEntriesWidget::ToDoEntriesWidget()
+{
+    set_layout<GUI::VerticalBoxLayout>();
+    m_result_view = add<GUI::TableView>();
+
+    m_result_view->on_activation = [](auto& index) {
+        auto& match = *(const Cpp::TodoEntry*)index.internal_data();
+        open_file(match.filename);
+        current_editor().set_selection({ { match.line, match.column }, { match.line, match.column } });
+        current_editor().set_focus(true);
```

Comment:
I think using the version of `open_file` that takes a line&column would be better here.
(Also, I don't think we really need to have the TODO line selected, just jumping to it in the source is enough).
```suggestion
        open_file(match.filename, match.line, match.column);
```

Reply:
Done :)